core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -394,6 +394,27 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
   }
 
   /**
+   * Request that the cluster manager kill the specified executor without adjusting the
+   * application resource requirements.
+   *
+   * The effect is that a new executor will be launched in place of the one killed by
+   * this request. This assumes the cluster manager will automatically and eventually
+   * fulfill all missing application resource requests.
+   *
+   * Return whether the request is received.
+   */
+  def killAndReplaceExecutor(executorId: String): Boolean = synchronized {
```

Comment:
Perhaps instead of adding this method, add a boolean flag to `killExecutors` above?

(Or, because of `ExecutorAllocationClient`, create a private method that takes the list of executors and a boolean saying whether to update the target executor count, and call it from both of these methods.)